appserver/monitoring-console/core/src/main/java/fish/payara/monitoring/model/Series.java

```diff
@@ -52,8 +52,8 @@
 public final class Series implements Comparable<Series>, Serializable {
 
     public static final char TAG_ASSIGN = ':';
-    public static final char TAG_SEPARATOR = ' ';
-    private static final char[] TAG_SEPARATORS = { ' ', ',', ';' };
+    public static final char TAG_SEPARATOR = '\u001F'; //As tag values can be any valid UTF-8 character, use a control character to separate tags
```

Comment:
The design that series are human readable and writable is intentional - using a control character compromises this property which I don't like as it messes up tooling and ease of testing things manually. The benefit of using a control character would only fully apply if the `TAG_ASSIGN` is made some control character as well. Besides that it is superficial because there would just be other characters that mess up the parsing even if we can expect them to be less frequent. 

We have no requirement to support unchanged arbitrary names so I think the solution I suggested to simply append the tag name with special character replacement is the most balanced. It yields the advantages of a clear human readable and writable format which are useful in many regards while its disadvantages of character replacement only apply to edge cases where a replacement does not alter the name or value in a way that would make it hard to recognise.

Reply:
If this is meant to be human-readable, that is what MetricWriter and its sub-classes are for. The PR includes a test which ensures that odd values are written correctly, so TAG_ASSIGN does not need to be a control character.

Reply:
I don't say that the conversion of metric data is how it should be. Still, the collection API should protect against values being passed that later on fail to parse to a `Series` object that expects certain characters to have special meaning. I opted to simply replace those characters with `_` as all of them are kind of a separator character which `_` can replace quite ok.
So if the collection API has this mechanism anyway there is little point to use custom `MetricWriter`s unless we really want to have them map to other replacement characters. 

Anyhow, the pain that would be inflicted by a non human readable and writable format would be far far bigger then to little itch of having 5 characters replaced with `_`. You have to remember that the series name is THE identifier concept that humans do read and write, copy & paste, send in mails and might even put in URLs. Compare the headache a control character in that would cause in comparison to the minor inconvenience some replaced characters in odd names have.